resources/knative-eventing/charts/knative-eventing/templates/pre-upgrade-job.yaml

```diff
@@ -15,10 +15,21 @@ data:
     #!/usr/bin/env bash
     set -eu
     echo "*** Pre upgrade job starts ***"
-    kubectl scale -n kyma-system deploy/event-bus-subscription-controller-knative --replicas=0
+  
+    kubectl delete deploy/event-bus-subscription-controller-knative -n kyma-system
```

Comment:
Now, this may cause problem due to this [issue](https://github.com/kyma-project/kyma/issues/4989). To be more resilient let's add `--ignore-not-found` to each delete line. And you can reduce code by:
```
kubectl delete svc,deploy -l app=publish-knative -n kyma-system
kubectl delete svc,deploy -l app=subscription-controller-knative -n kyma-system
```

We should delete Clusterroles, Clusterrolebinding, ServiceAccounts also. E.g. `event-bus-publish-knative` `event-bus-subscription-controller-knative-sa`

Reply:
ServiceAccount cleanup introduced, however w.r.t CLusterRoles and Bndings no chang has been introduced and i intend to keep it that way as it's name justifies the role